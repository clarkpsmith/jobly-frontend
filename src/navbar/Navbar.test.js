import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";
import UserContext from "../common/UserContext";

const currentUser = {
  applications: [
    200,
    161,
    86,
    50,
    71,
    158,
    36,
    134,
    78,
    184,
    60,
    181,
    89,
    172,
    62,
    82,
    98,
    47,
    183,
    29,
    63,
    53,
    114,
    69,
    7,
    150,
    18,
    95,
    177,
    23,
    123,
    175,
    100,
  ],
  email: "clark.smith79@gmail.com",
  firstName: "Clark",
  isAdmin: false,
  lastName: "Smith",
  username: "testuser",
};

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out ", () => {
  const currentUser = undefined;
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
