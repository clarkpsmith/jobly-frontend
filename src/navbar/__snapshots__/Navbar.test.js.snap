// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="navlinks"
            href="/companies"
          >
            Companies
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="navlinks"
            href="/jobs"
          >
            Jobs
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="navlinks"
            href="/profile"
          >
            Profile
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="navlinks-logout active"
            href="/"
          >
            Log Out testuser
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out  1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="navlinks"
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="navlinks"
            href="/signup"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
